SPECS/setools/setools.spec

```diff
@@ -0,0 +1,185 @@
+%global selinux_ver 2.9-1
+%global __python3	/usr/bin/python3
+%define python3_sitearch %(python3 -c "from distutils.sysconfig import get_python_lib; import sys; sys.stdout.write(get_python_lib(1))")
+
+Name:           setools
+Version:        4.2.2
+Release:        2%{?setools_pre_ver:.%{setools_pre_ver}}%{?dist}
+Summary:        Policy analysis tools for SELinux
+# binaries are GPL and libraries are LGPL.  See COPYING.
+License:        GPLv2 and LGPLv2+
+URL:            https://github.com/SELinuxProject/setools
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Source0:        https://github.com/SELinuxProject/setools/releases/download/%{version}/%{name}-%{version}.tar.bz2
+
+BuildRequires: flex
+BuildRequires: bison
+BuildRequires: glibc-devel
+BuildRequires: gcc
+BuildRequires: git
+BuildRequires: libsepol-devel >= 2.9-1
+BuildRequires: qt5-qtbase-devel
+BuildRequires: swig
+BuildRequires: python3-Cython
+BuildRequires: python3-devel
+BuildRequires: python3-setuptools
+BuildRequires: python3-xml
+BuildRequires: libselinux-devel
+
+%description
+SETools is a collection of graphical tools, command-line tools, and
+Python modules designed to facilitate SELinux policy analysis.
+
+%package     console
+Summary:     Policy analysis command-line tools for SELinux
+License:     GPLv2
+Requires:    setools-python3 = %{version}-%{release}
+Requires:    libselinux >= %{selinux_ver}
+
+%description console
+SETools is a collection of graphical tools, command-line tools, and
+libraries designed to facilitate SELinux policy analysis.
+
+This package includes the following console tools:
+
+  sediff       Compare two policies to find differences.
+  seinfo       List policy components.
+  sesearch     Search rules (allow, type_transition, etc.)
+
+%package     python3
+Summary:     Policy analysis tools for SELinux
+Obsoletes:   setools-libs < 4.0.0
+Recommends:  libselinux-python3
+Requires:    python3-setuptools
+
+%description python3
+SETools is a collection of graphical tools, command-line tools, and
+Python 3 modules designed to facilitate SELinux policy analysis.
+%prep
+%setup -n %{name}
+
+%build
+pwd
+%{__python3} setup.py build_ext
+%{__python3} setup.py build
+
+
+%install
+%{__python3} setup.py install --prefix=%{_prefix} --root=%{buildroot}
+
+# Remove unpackaged files.  These are tools for which the dependencies
+# are not yet available on mariner (python3-networkx)
+rm -rf %{buildroot}/%{_bindir}/sedta
+rm -rf %{buildroot}/%{_bindir}/seinfoflow
+rm -rf %{buildroot}/%{_mandir}/man1/sedta*
+rm -rf %{buildroot}/%{_mandir}/man1/seinfoflow*
+rm -rf %{buildroot}/%{_bindir}/apol
+rm -rf %{buildroot}/%{python3_sitearch}/setoolsgui
+rm -rf %{buildroot}/%{_mandir}/man1/apol*
+
+%files
+%license COPYING COPYING.GPL COPYING.LGPL
```

Comment:
Curious: do you know why the default `setools` is even being built? It looks like it has no runtime dependencies, not pre/post-install steps and it installs no files. Is it possible we can just remove thie one `%file` section?

Reply:
That was inherited from Fedora.  I'm not sure exactly how it works there.  When I run "dnf deplist setools" on my Fedora box, it lists several dependencies (to other packages in the same spec, such as setools-console), but I don't see them in the Fedora version of the spec.  I don't understand spec files well enough to know how that works.  Does it happen implicitly somehow?

In our context, removing the %file section as you suggest seems fine to me.  I'll make that change.

Reply:
Thank you.

And to answer your question: if there are no explicitly mentioned dependencies in the spec between the default package and one of the subpackages, then I'm also a bit confused, sorry.

---
In reply to: [487020894](https://github.com/microsoft/CBL-Mariner/pull/100#discussion_r487020894) [](ancestors = 487020894)